src/application.py

```diff
@@ -12,6 +12,10 @@
 
 @asynccontextmanager
 async def lifespan(application: FastAPI):
+    """
+    lifespan function that yields a connection to the database that lasts until the code is shut down.
+    :param application: FastAPI object, the app.
+    """
     # At startup - start connection to the SQL server
     connection_manager = get_db()
     application.state.connection_manager = connection_manager
```

Comment:
I'd suggest to wrap this in a try-except statement with a few retries. Sometimes when you spin up services together, your app service might start before your database is even ready so the app startup may fail. Also, make sure you give some time before each retry attempt. :)